backend/Dockerfiles/Dockerfile.veracode

```diff
@@ -76,28 +86,32 @@ RUN mkdir -p /node && \
     mv /node/node-v$NODEVER-linux-x64 /node/node && \
     rm /node/node.tar.gz
 
-FROM python:3.9-bullseye as ruby-builder
+FROM php@${PHP_IMG_VER} as php-builder
 
-ARG RUBYVER=2.7.2
-ARG RUBYSHA=3f50b100fb52cdf315fa17f41ae2e2538bb0c45abd9a6c569fd70ac851d61b2a
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
-RUN mkdir -p /ruby && \
-    echo "$RUBYSHA /ruby/ruby.tar.bz" > /ruby_checksum.txt && \
-    curl https://rvm_io.global.ssl.fastly.net/binaries/debian/10/x86_64/ruby-$RUBYVER.tar.bz2 -L -o /ruby/ruby.tar.bz && \
-    sha256sum -c /ruby_checksum.txt && \
-    tar -xjvf /ruby/ruby.tar.bz -C /ruby && \
-    mv /ruby/ruby-$RUBYVER /ruby/ruby && \
-    rm /ruby/ruby.tar.bz
+RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
 
-FROM php:7.4.22-cli-bullseye as php-builder
+FROM python@${PYTHON_IMG_VER} as java-builder
 
-RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+ARG JAVAVER=17.0.8
+ARG JAVASHA=74b528a33bb2dfa02b4d74a0d66c9aff52e4f52924ce23a62d7f9eb1a6744657
+
+RUN mkdir -p /java && \
+    echo "$JAVASHA java.tar.gz" >java_checksum.txt && \
+		JAVAMAJOR=$(echo "${JAVAVER}" | cut -d . -f 1) && \
+    curl "https://download.oracle.com/java/${JAVAMAJOR}/archive/jdk-${JAVAVER}_linux-x64_bin.tar.gz" -L -o java.tar.gz && \
+    sha256sum -c java_checksum.txt && \
+    tar -xzvf java.tar.gz --strip-components 1 -C /java
```

Comment:
Had to install java from Oracle as opposed to apt. There seems to be a bug in the post-installation script of the deb package which causes installation to fail.